### ISSUE TITLE
Add text helpers

### ISSUE THREAD

**Issue by ianmackenzie** [ianmackenzie/elm-2d-drawing#7]
Nicer interface to SVG `<tspan>`, `dy` etc. - maybe support Markdown or something?

**Comment by gampleman** [comment]
So I think there are 2 pretty common use cases for text in SVG:

1. Labelling a drawing or diagram.
2. [Fancy text effects](https://codepen.io/rgg/pen/ozLzbz).

I think doing richly formatted text is actually pretty rare, since it's better to just embed HTML for that. But the main problem with 1 is that often you do not exactly know the dimensions that the text will take and so you need to estimate those and break your text manually into multiple lines.

For example, in one project I have some code that looks a bit like this:

```elm
makeLabel : (Float, Float) -> TextAlignment -> List String -> Svg msg
makeLabel position align lines =
    text_
           [ y (Tuple.second position - spaceAboveGuideLine)
            , x (Tuple.first position)
            , textAnchor align
            , fill "var(--foreground-color)"
            ]
            (case lines of
                  [ head ] ->
                       [ text head ]
                  
                  _ ->
                       List.indexedMap (\index word -> 
                           tspan 
                              [ dy (String.fromInt (if index == 0 then 1 - List.length lines else 1)) ++ "em"
                              , x (Tuple.first position)] 
                              [ text word ]
                      )

             )
```

with some additional logic that splits strings into lines based on visual width. Obviously figuring out that logic took a good chunk of time and ended up entangled with a lot of the other logic needed to produce that specific image.

**Comment by ianmackenzie** [comment]
Yeah, something like `makeLabel` could be good - maybe have something like

```elm
Drawing2d.text : List Attribute -> Point2d -> String
Drawing2d.paragraph : List Attribute -> Point2d -> List String
```

and the second one would basically be your `makeLabel` function (I've omitted some type variables for simplicity). It might even be possible to be clever and adjust text alignment based on the specified anchor position (given in the attributes) - e.g. if the anchor is set to bottom right, then the `<text>` entity is created starting from the *last* line (with text anchor set to bottom right) and then other lines are offset upwards (and right-aligned somehow - not sure exactly how that works).

You'd still have to figure out line breaking yourself - to me that seems like something that would make more sense to implement in a separate package if you wanted to do it automatically, since there's a lot of complexity there and it's not specific to 2D drawings.

**Comment by gampleman** [comment]
Yeah that makes sense. I think @folkertdev already has a package for doing that exactly, and I've built a very [simple heuristic solution](https://observablehq.com/@gampleman/fast-string-size-estimator) for doing it very quickly. So I think there are already pretty good solutions for that.
